### PR TITLE
Add github action for publishing

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://jda.wiki/contributing/contributing/

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [x] Other: CI and publishing

Closes Issue: NaN

## Description

This adds a new github action we can use to publish releases manually through github instead of jenkins.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -66,10 +66,17 @@ val previousVersion: Version by lazy {
         versionObj
 }
 
-val isNewVersion = previousVersion != versionObj
+val ossrhConfigured = getProjectProperty("ossrhUser") != null
+val canSign = getProjectProperty("signing.keyId") != null
```
Comment (Andre601) [inline build.gradle.kts:70]: Shouldn't this be `signingKeyId` given it is defined as such in the environment vars of GitHub Action? Or does it get this info from somewhere else?

Comment (MinnDevelopment) [reply build.gradle.kts]: Yes, I forgot to change the key logic for github actions. Thanks for pointing this out!

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          java-version: 17
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+      - name: Release to Maven Central
+        env:
+          ORG_GRADLE_PROJECT_ossrhUser: ${{ secrets.OSSRH_USER }}
+          ORG_GRADLE_PROJECT_ossrhPassword: ${{ secrets.OSSRH_PASSWORD }}
+          ORG_GRADLE_PROJECT_stagingProfile: ${{ secrets.STAGING_PROFILE_ID }}
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_KEY }}
+          ORG_GRADLE_PROJECT_signingKeyId: ${{ secrets.GPG_KEYID }}
```
Comment (DV8FromTheWorld) [inline .github/workflows/publish.yml:29]: are these keys already setup? or do we still need to?

Comment (MinnDevelopment) [reply .github/workflows/publish.yml:29]: They are setup here https://github.com/discord-jda/JDA/settings/environments/4057422970/edit

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: "publish"
+
+jobs:
+  publish:
+    environment: MavenRelease
```
Comment (MinnDevelopment) [inline .github/workflows/publish.yml:11]:
```suggestion
    environment: maven-release
```

**Pull request merged**